frontend/src/views/StatblockEditorView.vue

```diff
@@ -686,6 +705,155 @@ export default defineComponent({
+					offenseCR = cr
+				}
+			})
+
+			
+			const attackBonusDiff = this.crCalc.attackBonus - ChallengeRatingTable[offenseCR].attackBonus
+			const saveDiff = this.crCalc.dc - ChallengeRatingTable[offenseCR].dc 
+			
+
+			defenseCR += Math.floor((this.data.defenses.ac.ac - ChallengeRatingTable[defenseCR].ac)/2.0)
+			offenseCR += Math.floor(Math.max(attackBonusDiff, saveDiff) / 2.0)
+
+			const totalCR = (defenseCR: number, offenseCR: number): number => {
+				const avg = (defenseCR + offenseCR) / 2
+
+				if (avg >= 0.75) return Math.min(30, Math.round(avg));
+				if (avg >= 0.375) return 0.5;
+				if (avg >= 0.1875) return 0.25;
+				if (avg >= 0.0625) return 0.125;
+				return 0
+			}
+			
+			this.showCRModal = false
+			this.data.description.cr = totalCR(defenseCR, offenseCR)
+			toast.success("Defense CR: " + defenseCR + " Offense CR: " + offenseCR + " Total Calcualted CR: " + totalCR(defenseCR, offenseCR))
+
+		},
+		hpCalc(): number {
+            return Math.floor(this.data.defenses.hp.numOfHitDie * ( (this.data.defenses.hp.sizeOfHitDie + 1)/2 + this.statCalc("con")))
+        },
+        statCalc(stat: Stat) : number {
+            return Math.floor(this.data.abilities.stats[stat]/2)-5
+        },
+		attackStats(): number[]{
+			var bonuses: number[] = []
+			var saveDC: number[] = []
+			var damages: number[] = []
+
+			function findAttackBonus(obj: any): any{
+				if (typeof obj === 'object' && obj !== null){
+					if ('attackBonus' in obj){
+						return parseInt(obj.attackBonus)
+					} else {
+						for (const key in obj){
+							const result = findAttackBonus(obj[key])
+							if (result !== undefined){
+								return result
+							}
+						}
+					}
+				}
+			}
+
+			function findAttackDamage(obj: any): any{
+				if (typeof obj === 'object' && obj !== null){
+					if ('damage' in obj){
+						return averageDamage(obj.damage)
+					} else {
+						for (const key in obj){
+							const result = findAttackDamage(obj[key])
+							if (result !== undefined){
+								return result
+							}
+						}
+					}
+				}
+			}
+
+			function findDC(obj: any): any{
+				if (typeof obj === 'object' && obj !== null){
+					if ('dc' in obj){
+						return obj.dc
+					} else {
+						for (const key in obj){
+							const result = findDC(obj[key])
+							if (result !== undefined){
+								return result
+							}
+						}
+					}
+				}
+			}
+
+			function averageDamage(damageString: string): number{
+				const d_string = damageString.replace(/\s*\[.*?\]\s*/g, '')
+				const [roll, mod] = d_string.includes("+")  ? d_string.split("+") : [d_string,0]
+				const [numDie, dieSize] = roll.split("d").map(Number)
+				const averageRoll = (dieSize+1)/2
+				const averageDmg = (averageRoll * numDie) + (mod ? Number(mod) : 0)
+				return Math.floor(averageDmg)
+			}
+
+			function scrapeFeatures(features: FeatureEntity[]){
+				features.forEach(feature => {
+					var automation: string
+
+					if (feature.automation){
```

Comment:
here you are stringifying some JSON and then immediately parsing them back into JSON again later. This is a performance concern. If it isn't parsed from the description, it shouldn't be stringified and parsed